Upgrade the base OS for GH runner to openSUSE 15.6

Since yesterday the GH actions have been failing, possibly due to outdated repository used by the runner. The base OS for GH runners is openSUSE 15.4, updating the image to 15.5 might fix this problem.

<details><summary><code>sudo zypper install -y azure-cli</code></summary>
<p>

```sh
Retrieving repository 'Update repository of openSUSE Backports' metadata [...
Warning: The gpg key signing file 'repomd.xml' has expired.
  Repository:       Update repository of openSUSE Backports
  Key Fingerprint:  637B 32FF 3D83 F07A 7AE1 C40A 9C21 4D40 6517 6565
  Key Name:         openSUSE:Backports OBS Project <openSUSE:Backports@build.opensuse.org>
  Key Algorithm:    RSA 2048
  Key Created:      Fri 26 Nov 2021 02:26:23 PM UTC
  Key Expires:      Sun 04 Feb 2024 02:26:23 PM UTC (EXPIRED)
  Rpm Name:         gpg-pubkey-65176565-61a0ee8f
......done]
Building repository 'Update repository of openSUSE Backports' cache [....done]
Problem retrieving files from 'Debug Repository'.
Location 'http://download.opensuse.org/debug/distribution/leap/15.4/repo/oss/media.1/media' is temporarily unaccessible.
Warning: Skipping repository 'Debug Repository' because of the above error.
Please see the above error message for a hint.
Retrieving repository 'Update Repository (Debug)' metadata [.....done]
Building repository 'Update Repository (Debug)' cache [....done]
Retrieving repository 'Non-OSS Repository' metadata [...done]
Building repository 'Non-OSS Repository' cache [....done]
Retrieving repository 'Main Repository' metadata [..done]
Building repository 'Main Repository' cache [....done]
Retrieving repository 'Update repository with updates from SUSE Linux Enterprise 15' metadata [...............................................done]
Building repository 'Update repository with updates from SUSE Linux Enterprise 15' cache [....done]
Problem retrieving files from 'Source Repository'.
Location 'http://download.opensuse.org/source/distribution/leap/15.4/repo/oss/media.1/media' is temporarily unaccessible.
Warning: Skipping repository 'Source Repository' because of the above error.
Please see the above error message for a hint.
Retrieving repository 'Main Update Repository' metadata [........done]
Building repository 'Main Update Repository' cache [....done]
Retrieving repository 'Update Repository (Non-Oss)' metadata [....done]
Some of the repositories have not been refreshed because of an error.
Building repository 'Update Repository (Non-Oss)' cache [....done]
Loading repository data...
Warning: Repository 'Update repository of openSUSE Backports' appears to be outdated. Consider using a different mirror or server.
Reading installed packages...
Resolving package dependencies...
...
...
...
(148/149) Installing: azure-cli-core-2.17.1-150100.6.18.1.noarch [............done]
(149/149) Installing: azure-cli-2.17.1-150100.6.11.2.noarch [............done]
Error: Process completed with exit code 106.
```
Ref: https://github.com/rancher/hosted-providers-e2e/actions/runs/7794625205/job/21256289270
</p>
</details> 

It appears to be fixed by changes in https://github.com/rancher/hosted-providers-e2e/tree/improve_gha, but it is still better to upgrade to 15.5.

GH Actions:
1. https://github.com/rancher/hosted-providers-e2e/actions/runs/7794625205/job/21256289270
2. [AKS-E2E](https://github.com/rancher/hosted-providers-e2e/actions/runs/7794648805)
3. [GKE-E2E](https://github.com/rancher/hosted-providers-e2e/actions/runs/7794625205)
4. [EKS-E2E](https://github.com/rancher/hosted-providers-e2e/actions/runs/7794593497)

Things to note after upgrading to 15.5:
1. We will need to add 2 additional steps to the workflow to avoid problems such as `XXX binary not found`:
```github-action
    steps:
        - name: Adding /usr/local/bin into PATH
          run: echo "/usr/local/bin/" >> ${GITHUB_PATH}
        - name: Install Node
          uses: actions/setup-node@v4
          with:
            node-version: 16
```
---
07/24 EDIT: I decided to bump directly to openSUSE 15.6 in the end. I have preinstalled devel_python3 pattern, npm (so the setup-node is not needed) and azure-cli (it should be just updated during a run) to speed up test provisioning.

--- a/.github/workflows/aks.yaml
+++ b/.github/workflows/aks.yaml
@@ -19,7 +19,7 @@ on:
         default: v1.28.9+k3s1
       runner_template:
         description: Runner template to use
-        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1
+        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2
         type: string
       operator_nightly_chart:
         description: Install hosted-provider nightly chart
@@ -57,7 +57,7 @@ jobs:
       operator_nightly_chart: ${{ inputs.operator_nightly_chart == true || (github.event_name == 'schedule' && true) }}
       tests_to_run: ${{ inputs.tests_to_run || 'p0_provisioning/p0_importing' }}
       destroy_runner: ${{ inputs.destroy_runner == true || (github.event_name == 'schedule' && true) }}
-      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1' }}
+      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2' }}
       downstream_k8s_minor_version: ${{ inputs.downstream_k8s_minor_version }}
       rancher_installed: ${{ inputs.rancher_installed || 'hostname/password' }}
       downstream_cluster_cleanup: ${{ inputs.downstream_cluster_cleanup == true || (github.event_name == 'schedule' && true) }}

--- a/.github/workflows/eks.yaml
+++ b/.github/workflows/eks.yaml
@@ -19,7 +19,7 @@ on:
         default: v1.28.9+k3s1
       runner_template:
         description: Runner template to use
-        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1
+        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2
         type: string
       operator_nightly_chart:
         description: Install hosted-provider nightly chart
@@ -57,7 +57,7 @@ jobs:
       operator_nightly_chart: ${{ inputs.operator_nightly_chart == true || (github.event_name == 'schedule' && true) }}
       tests_to_run: ${{ inputs.tests_to_run || 'p0_provisioning/p0_importing' }}
       destroy_runner: ${{ inputs.destroy_runner ==true || (github.event_name == 'schedule' && true) }}
-      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1' }}
+      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2' }}
       downstream_k8s_minor_version: ${{ inputs.downstream_k8s_minor_version }}
       rancher_installed: ${{ inputs.rancher_installed || 'hostname/password' }}
       downstream_cluster_cleanup: ${{ inputs.downstream_cluster_cleanup == true || (github.event_name == 'schedule' && true) }}

--- a/.github/workflows/gke.yaml
+++ b/.github/workflows/gke.yaml
@@ -19,7 +19,7 @@ on:
         default: v1.28.9+k3s1
       runner_template:
         description: Runner template to use
-        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1
+        default: hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2
         type: string
       operator_nightly_chart:
         description: Install hosted-provider nightly chart
@@ -57,7 +57,7 @@ jobs:
       operator_nightly_chart: ${{ inputs.operator_nightly_chart == true || (github.event_name == 'schedule' && true) }}
       tests_to_run: ${{ inputs.tests_to_run || 'p0_provisioning/p0_importing' }}
       destroy_runner: ${{ inputs.destroy_runner ==true || (github.event_name == 'schedule' && true) }}
-      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v1' }}
+      runner_template: ${{ inputs.runner_template || 'hosted-prov-e2e-ci-runner-spot-n2-highmem-16-gl-template-v2' }}
       downstream_k8s_minor_version: ${{ inputs.downstream_k8s_minor_version }}
       rancher_installed: ${{ inputs.rancher_installed || 'hostname/password' }}
       downstream_cluster_cleanup: ${{ inputs.downstream_cluster_cleanup == true || (github.event_name == 'schedule' && true) }}

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -108,6 +108,11 @@ jobs:
     outputs:
       PUBLIC_IP: ${{ steps.runner-ip.outputs.PUBLIC_IP }}
     steps:
+      - name: Add /usr/local/bin into PATH
+        run: |
+          echo "/usr/local/bin/" >> ${GITHUB_PATH}
+          echo 'Defaults secure_path="/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/bin"' | sudo tee /etc/sudoers.d/0-custom_secure_path
+
       - name: Checkout
         uses: actions/checkout@v4
 

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -105,6 +105,9 @@ jobs:
   installation-and-e2e-tests:
     runs-on: ${{ needs.create-runner.outputs.uuid }}
     needs: create-runner
+    env:
+      # For some reason the go doesn't link by default against system libresolv library
+      CGO_LDFLAGS: -O2 -g -lresolv
     outputs:
       PUBLIC_IP: ${{ steps.runner-ip.outputs.PUBLIC_IP }}
     steps:

